lib/services.js

```diff
@@ -377,47 +377,28 @@ export default {
             log.trace('object identified as non-versioned');
             // non-versioned buckets
             log.trace('deleteObject: deleting non-versioned object');
-            if (objectMD.location === null) {
-                return metadata.deleteObjectMD(bucketName, objectKey, log, cb);
-            } else if (!Array.isArray(objectMD.location)) {
-                return data.delete(objectMD.location, log, err => {
+            return metadata.deleteObjectMD(bucketName, objectKey, log,
+                err => {
                     if (err) {
-                        log.error('error deleting from dataStore', {
-                            error: err,
-                        });
                         return cb(err);
                     }
-                    log.trace('deleteobject: data delete ok');
-                    metadata.deleteObjectMD(bucketName, objectKey, log, cb);
-                    return undefined;
-                });
-            }
-            const errs = [];
-            async.eachLimit(objectMD.location, 5, (loc, ok) => {
-                data.delete(loc, log, err => {
-                    if (err) {
-                        errs.push(err);
-                        log.error('error deleting from dataStore', {
-                            error: err,
-                        });
+                    // TODO: The method of persistence of sproxy delete key will
+                    // be finalized; refer Issue #312 for the discussion. In the
+                    // meantime, we at least log the location of the data we are
+                    // about to delete before attempting its deletion.
+                    log.trace('deleteObject: metadata delete OK; about to ' +
+                    'delete object in data', { location: objectMD.location });
+                    if (objectMD.location === null) {
+                        return cb();
+                    } else if (!Array.isArray(objectMD.location)) {
+                        return data.delete(objectMD.location, log, cb);
```

Comment:
Since we don't really care if data deletion fails at the moment, no (or not harmful) callback in `data.delete` is necessary. Or you can make a loop retrying (with limit) deleting the stubborn parts.